### ISSUE TITLE
Raise error if an attribute on a coroutine is accessed it doesn't have

### ISSUE THREAD

**Issue by Martmists-GH** [pylint-dev/pylint#1496]
Some people forget to use `await`, so it might be useful to error in cases such as this:

```py
async def my_coro(*args):
    result = other_coro(*args)  # whoops, we didn't await
    x = result.some_attr  # error, coroutines don't have the some_attr attribute
    y = result["key"]  # error, coroutines don't have keys. The same applies to result[0]
    return result  # valid, coroutines can be returned as result sometimes.
```
I'd love to see this implemented in pylint at least for coroutines, and maybe for other awaitables too.

**Comment by PCManticore** [comment]
Right, this might be interesting to have

**Comment by mbyrnepr2** [comment]
A more recent report for this exists at https://github.com/pylint-dev/pylint/issues/8870. Let's close this and add the example to the more recent report.
